tests/test_extension_base.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+
+from syrupy.extensions.base import SnapshotReporter
+
+from .utils import clean_output
+
+
+class SnapshotReporterNoContext(SnapshotReporter):
+    @property
+    def _context_line_count(self) -> int:
+        return 0
+
+
+@pytest.mark.parametrize("Reporter", [SnapshotReporter, SnapshotReporterNoContext])
+class TestSnapshotReporter:
+    @pytest.mark.parametrize(
+        "a, b",
+        [
+            (
+                "line 0\nline 1\nline 02\nline 3\nline 4\r\nline 5\nline 6\nline 7",
+                "line 0\nline 1\nline 2\r\nline 3\nline 04\nline 5\nline 6\nline 7",
```

Comment:
more cases:
- trailing newlines
- where the difference between 2 sets is extra whitespace (space, tab) at the end
- different number of lines (additions + deletions)